docs/built-in-functions.rst

```diff
@@ -664,6 +664,25 @@ Math
         >>> ExampleContract.foo(9.0)
         3.0
 
+.. py:function:: isqrt(x: uint256) -> uint256
+
+    Return the square root of the provided integer number, using Babylonian square root algorithm.
+
+    .. note::
+        Performance note: If ``x+1`` is a perfect square, then the Babylonian algorithm oscillates between floor(sqrt(x)) and ceil(sqrt(x)) in consecutive iterations. ``isqrt`` has a final check that returns the floor value always, but this increases costs by approximately 10%.
```

Comment:
can the user configure this? adding this as a performance note makes it sounds like they can -- this comment seems more directed at future maintainers of the implementation, in which case it would make more sense as comments in the implementation source code.

Reply:
a user can configure this indeed. we could set up a `precise = True` flag in the `isqrt` builtin if you think that helps users shave off some gas?

Reply:
I think being configurable is not worth it at this point and we should move the note to the source code (for future maintainers).